UltraStar Play/Assets/Common/UI/Dialogs/DialogBackdrop.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class DialogBackdrop : MonoBehaviour
+{
+    public Image background;
+    public Color targetColor = new Color32(0, 0, 0, 90);
+    public float colorTransitionTime = 1f;
+
+    private float startTime;
+    private Color startColor;
+    
+    public void Start()
+    {
+        startTime = Time.time;
+        startColor = background.color;
+        
+        // Make the background match the screen size, regardless of its parent.
+        RectTransform backgroundTransform = background.GetComponent<RectTransform>();
+        backgroundTransform.position = new Vector2(0, 0);
+        backgroundTransform.anchorMax = new Vector2(Screen.width, Screen.height);
```

Comment:
anchorMax is relative to parent, where 1 is all of parent width/height.
Setting it to Screen.width would be 1000 times the parent width, but not really independent of the parent.
Furthermore, the anchor is not automatically applied to the corners, so I wonder why this works (if it does for you).

Probably better:
rectTransform.SetSizeWithCurrentAnchors(RectTransform.Axis.Horizontal, Screen.width);
rectTransform.SetSizeWithCurrentAnchors(RectTransform.Axis.Vertical, Screen.height);

Reply:
Oops, I feel kind of embarrassed for not noticing. You are correct, the background was scaled to about 1000 times the dialog (which "works" for filling the screen, as long as you don't look at it in scene mode ^^).

Fixed the issue. Also added a check to reapply the resize if the screen size changes while a dialog is open, because that could otherwise cause future issues with for example mobile device screen orientation changes.